KomaMRIBase/src/datatypes/Scanner.jl

```diff
@@ -28,19 +73,73 @@ julia> sys = Scanner()
 julia> sys.B0
 ```
 """
-@with_kw mutable struct Scanner
-    #Main
-    B0::Real=1.5
-    B1::Real=10e-6
-    Gmax::Real=60e-3
-    Smax::Real=500
-    #Sampling
-    ADC_Δt::Real=2e-6
-    seq_Δt::Real=1e-5
-    GR_Δt::Real=1e-5
-    RF_Δt::Real=1e-6
-    #Secondary
-    RF_ring_down_T::Real=20e-6
-    RF_dead_time_T::Real=100e-6
-    ADC_dead_time_T::Real=10e-6
+@with_kw mutable struct Scanner{T, GradType <: Gradients{T}, RFType <: RFCoils{T}}
+    limits::HardwareLimits{T} = HardwareLimits{Float64}()
+    gradients::GradType = LinearXYZGradients{Float64}()
+    rf_coils::RFType = UniformRFCoils{Float64}()
+ end
+
+function Base.view(sys::Scanner{T, GradType, RFType}, p) where {T, GradType <: Gradients{T}, RFType <: RFCoils{T}}
+    return sys
+ end
+
+ function Base.view(sys::Scanner{T, GradType, RFType}, p) where {T, GradType <: Gradients{T}, RFType <: ArbitraryRFCoils}
+    return sys
+ end
+
+ function Base.view(sys::Scanner{T, GradType, RFType}, p) where {T, GradType <: Gradients{T}, RFType <: RFCoilsSensDefinedAtPhantomPositions}
+    return Scanner(limits=sys.limits, gradients=sys.gradients, rf_coils=@view(sys.rf_coils[p]))
+ end
+
+ function Base.view(rf_coils::RFCoilsSensDefinedAtPhantomPositions, p)
+    return RFCoilsSensDefinedAtPhantomPositions(@view rf_coils.coil_sens[p,:])
+end
+
+function acquire_signal!(sig, obj, rf_coils::UniformRFCoils, Mxy)
+    sig .= transpose(sum(Mxy; dims=1))
+    return nothing
+end
+
+function acquire_signal!(sig, obj, rf_coils::RFCoilsSensDefinedAtPhantomPositions, Mxy)
+    for i in 1:size(rf_coils.coil_sens, 2)
+        sig[:, i] .= transpose(sum(rf_coils.coil_sens[:, i] .* Mxy; dims=1))
+    end
+    return nothing
+end
+
+function acquire_signal!(sig, obj, rf_coils::ArbitraryRFCoils, Mxy)
+    interpolated_coil_sens = similar(rf_coils.coil_sens, length(obj.x), size(rf_coils.coil_sens, 4))
+    coords = collect(zip(obj.x, obj.y, obj.z)) 
```

Comment:
Remove `collect` this is very bad practice. Also `zip`ing the `obj.x`, etc. is not good.